.github/workflows/conformance-kind-proxy-daemonset.yaml

```diff
@@ -0,0 +1,141 @@
+name: ConformanceKindEnvoyDaemonSet
```

Comment:
What do you think about using GHA matrix, and just re-use the same ConformanceKind GHA? Basically, it's to reduce the maintenance effort.

Reply:
@sayboras 

thought about that as well. but when introducing this - cilium cli 0.14 (with required helm mode) wasn't released - and therefore the diff even bigger.

and even now it kind of depends on https://github.com/cilium/cilium/issues/25156. 

happy to combine them in an additional step - where we might want to consider other tests too (e.g. ingress & gateway conformance tests).